discovery/pkg/sync/time.go

```diff
@@ -0,0 +1,28 @@
+// Copyright © 2018 Heptio
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sync
+
+import "time"
+
+// This is used to allow for unit testing values that rely on `time.Now()`.
+// In order to write tests and expect certain values, it's nessesary to provide
+// a way to override the default values.
+type nowFuncT func() time.Time
+
+var nowFunc nowFuncT
```

Comment:
I think this will result in a nil-pointer deref outside of tests

Reply:
yeah my bad, I got overzealous, I restored the old impl.